Ballerina-Training/Ballerina-Forum-Application/back-end/bal-forum/endpoints.bal

```diff
@@ -0,0 +1,117 @@
+import ballerina/http;
+import ballerina/log;
+import ballerina/task;
+import ballerinax/googleapis.gmail;
+import ballerinax/mysql;
+import ballerinax/mysql.driver as _;
+import ballerinax/nats;
+
+# Database related configurations
+type DatabaseConfig record {|
+    # The database host
+    string host;
+    # The database user
+    string user;
+    # The database user password
+    string password;
+    # The database name
+    string database;
+    # The database port
+    int port;
+|};
+
+configurable DatabaseConfig dbConfig = ?;
+
+final mysql:Client forumDbClient = check new (...dbConfig);
+
+# Sentiment endpoint related configurations
+type SentimentEndpointConfig record {|
+    # The endpoint URL
+    string endpointUrl;
+    # The retry interval
+    decimal retryInterval;
+    # The authentication configuration
+    record {|
+        # The refresh URL
+        string refreshUrl;
+        # The client ID
+        string clientId;
+        # The client secret
+        string clientSecret;
+        # The refresh token
+        string refreshToken;
+    |} authConfig;
+|};
+
+configurable SentimentEndpointConfig sentimentEpConfig = ?;
+
+final http:Client sentimentClient = check new (sentimentEpConfig.endpointUrl,
+    config = {
+        retryConfig: {
+            interval: sentimentEpConfig.retryInterval
+        },
+        auth: {
+            ...sentimentEpConfig.authConfig,
+            clientConfig: {
+                secureSocket: {
+                    cert: "./resources/public.crt"
+                }
+            }
+        },
+        secureSocket: {
+            cert: "./resources/public.crt"
+        }
+    }
+);
+
+final nats:Client natsClient = check new (url = nats:DEFAULT_URL);
+
+# Gmail related configurations
+type GmailAuthConfig record {|
+    # The refresh URL
+    string refreshUrl = gmail:REFRESH_URL;
+    # The refresh token
+    string refreshToken;
+    # The client ID
+    string clientId;
+    # The client secret
+    string clientSecret;
+|};
+
+configurable GmailAuthConfig gmailAuthConfig = ?;
+
+configurable string adminGmail = ?;
+
+final gmail:Client gmailClient = check new ({auth: gmailAuthConfig});
+
+class EmailSenderTask {
+    *task:Job;
+    private final string recipientName;
+    private final string recipientEmail;
+
+    isolated function init(string recipientName, string recipientEmail) {
+        self.recipientName = recipientName;
+        self.recipientEmail = recipientEmail;
+    }
+
+    public function execute() {
+        string subject = "Welcome to Ballerina Forum";
```

Comment:
Should this be a constant?

Reply:
Yes, will make it a constant

Reply:
Fixed via [1eae7ab](https://github.com/ballerina-guides/training-content/pull/12/commits/1eae7abbc8d1d6bdd3a677b4fac348875ee53fb7)